src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -2702,9 +2713,176 @@ namespace SimAirServingZones {
 	}
 
 	void
+	SolveWaterCoilController(
+		bool const FirstHVACIteration,
+		int const AirLoopNum,
+		std::string const & CompName,
+		int & CompIndex,
+		std::string const & ControllerName,
+		int ControllerIndex,
+		bool const HXAssistedWaterCoil
+	)
+	{
+
+		// SUBROUTINE INFORMATION
+		//       AUTHOR:  Richard Raustad (FSEC)
+		//       DATE WRITTEN:  July 2017
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// This subroutine solves for the controllers in the specfied air loop OA system.
+
+		// METHODOLOGY EMPLOYED:
+		// For the specified primary air system:
+		// (1) the specific component in the OA system is simulated
+		// (2) The controllers and their actions are simulated
+		// (3) Steps 2 and 3 are repeated until the control criteria are satisfied
+		// (4) convergence is passed back to SolveAirLoopControllers via PrimaryAirSystem( AirLoopNum ).ControlConverged( ControllerIndex )
+
+		// REFERENCES: None
+
+		// Using/Aliasing
+		using namespace DataHVACControllers;
+		using HVACControllers::ManageControllers;
+		using General::CreateSysTimeIntervalString;
+		using WaterCoils::SimulateWaterCoilComponents;
+		using DataAirLoop::AirLoopControlInfo;
+		using HVACHXAssistedCoolingCoil::SimHXAssistedCoolingCoil;
+
+		// Locals
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+		// TRUE if first full HVAC iteration in an HVAC timestep
+		// DO loop index; there are 2 passes the 2nd is done only if mass balance fails
+		// Index of the air loop being simulated
+		// TRUE when primary air system & controllers simulation has converged;
+		// Max number of iterations performed by controllers across this air loop
+		// Aggregated number of iterations across all controllers on this air loop
+		// Total number of times SimAirLoopComponents() has been invoked
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+		// Maximum iterations of an air system/controllers simulation sequence
+		int const MaxIter( 50 );
+		static gio::Fmt fmtLD( "*" );
+
+		// SUBROUTINE LOCAL VARIABLE DEFINITIONS
+		// TRUE if controller supports speculative warm restart
+		bool AllowWarmRestartFlag;
```

Comment:
In the future, go ahead and initialize the variable right here where you declare them when possible.  No need to separate it out.